avocado/core/runner.py

```diff
@@ -468,6 +469,12 @@ def _iter_variants(template, mux):
         :raises ValueError: When variant and template declare params.
         """
         for variant, params in mux.itertests():
+            variant_name = ''
+            for x in params[0]:
+                path = x.path[len('/run'):]
+                variant_name += path.replace('/', '-')
+            # trim first '-'
+            variant_name = variant_name[1:]
```

Comment:
This is the core of this feature, and it defines how to transform the map the variant to something other than the incremental counter.

This is exactly the issue I mentioned earlier, about the variant IDs.  @ldoktor please join us on this.